src/main/scala/subsystem/RocketSubsystem.scala

```diff
@@ -124,7 +124,9 @@ trait HasRocketTiles extends HasTiles
 
     // 2. clint+plic conditionally crossing
     val periphIntNode = rocket.intInwardNode :=* rocket.crossIntIn
-    periphIntNode := clint.intnode                   // msip+mtip
+    p(CLINTKey).map { params =>
+      periphIntNode := clintOpt.get.intnode          // msip+mtip
```

Comment:
doesn't this not work if you don't have a clint? I seem to recall that these interrupts are somewhat hard-coded in how they are hooked up. Shouldn't you tie `msip` and `mtip` to 0 if clintOpt is None?

Reply:
See this comment: https://github.com/freechipsproject/rocket-chip/blob/20a88768560ca810c0dbe79f1a33604268d921c7/src/main/scala/tile/Interrupts.scala#L51

Reply:
The option unpacking can also be streamlined:
`clintOpt.foreach { periphIntNode := _.intnode }`

I share @mwachs5's  concern, so maybe something like:
`periphIntNode := clintOpt.map(_.intnode).getOrElse(UInt(0))`

Reply:
I don't think you can make an IntNode out of UInt(0). You need to make some sort of interrupt source that has 2 dummy MTIP and MSIP outputs.

Reply:
Oh also I like @hcook 's `foreach` version of this better. Much more readable.

Reply:
Would `foreach` version work correctly if there is more than 1 CLINT (by mistake)?

Reply:
the type of Clint is Option, not Seq, right?

Reply:
Yes

Reply:
so then it does the right thing